Bump node.js 24.7 in Dockerfile

zsaladin opened the pull request:
### Description

https://github.com/pyodide/pyodide/pull/5869#discussion_r2313728072

To use glob pattern in builtin Node.js test runner, it should be upgraded.

zsaladin commented:
@ryanking13 

I wrote a PR(#5878) about changing the test runner for unitest.
Do you want to merge #5878 to this PR?

ryanking13 commented:
@zsaladin 

No, I would like to see if the Node.js 24 works well with the existing unittests.

zsaladin commented:
@ryanking13 

Thanks for your suggestion.

I agreed with your approach, so I modified the code as you said, but it occurred an error below on dd1cee32ae0645916a41932a01f588df0bb08b0f

```
npm error Tag name must not be a valid SemVer range: 0.29.0-dev.0
```

It seems `--tag`  doesn't take a whole version itself.
I found [this](https://docs.npmjs.com/adding-dist-tags-to-packages#example)

```
npm publish --tag beta
```

I fixed it with indicating `dev` on aba123cdc7ecaac0b2ea8307e5ada15694f46d0a

ryanking13 commented:
> It seems --tag doesn't take a whole version itself.

Hmm, I see. Thanks. That command is for a dry run, so I think it's fine.
